pkg/namespacestore/namespacestore.go

```diff
@@ -557,12 +559,21 @@ func RunCreateGoogleCloudStorage(cmd *cobra.Command, args []string) {
 // RunCreateS3Compatible runs a CLI command
 func RunCreateS3Compatible(cmd *cobra.Command, args []string) {
 	createCommon(cmd, args, nbv1.NSStoreTypeS3Compatible, func(namespaceStore *nbv1.NamespaceStore, secret *corev1.Secret) {
+		log := util.Logger()
 		endpoint := util.GetFlagStringOrPrompt(cmd, "endpoint")
 		targetBucket := util.GetFlagStringOrPrompt(cmd, "target-bucket")
 		sigVer, _ := cmd.Flags().GetString("signature-version")
 		secretName, _ := cmd.Flags().GetString("secret-name")
 		mandatoryProperties := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"}
-
+		u, _ := url.Parse(endpoint)
+		if u.Scheme == "http" {
+			if sigVer == "v4" {
+				log.Fatalf("Non-secure endpoint works only with v2 signature-version. Please select signature version v2 for namespacestore")
+			}
+			sigVer = "v2"
+		} else {
+			sigVer = "v4"
```

Comment:
I think you need to set to "v4" only if the original is an empty string ("") no? In HTTPS, we support both versions, right?
```suggestion
			if sigVer == "" {
			       sigVer = "v4"
			}
```